### PR TITLE
ROX-20757: add multi-arch builds to konflux pipeline

### PR DESCRIPTION
**Pull request by Stringy**
## Description

A detailed explanation of the changes in your PR.

Feel free to remove this section if it is overkill for your PR, and the title of your PR is sufficiently descriptive.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Updated documentation accordingly

**Automated testing**
  - [ ] Added unit tests
  - [ ] Added integration tests
  - [ ] Added regression tests

If any of these don't apply, please comment below.

## Testing Performed

TODO(replace-me)
Use this space to explain how you tested your PR, or, if you didn't test it, why you did not do so. (Valid reasons include "CI is sufficient" or "No testable changes")
In addition to reviewing your code, reviewers **must** also review your testing instructions, and make sure they are sufficient.

For more details, ref the [Confluence page](https://stack-rox.atlassian.net/wiki/spaces/StackRox/pages/855998488/Proposal+Explicitly+List+Testing+Steps+on+PRs) about this section.

### CODE REVIEW

File: .tekton/collector-component-pipeline.yaml [hunk 1]
```diff
@@ -101,7 +101,7 @@ spec:
     value: $(tasks.clone-repository.results.commit)
   - description: ""
     name: JAVA_COMMUNITY_DEPENDENCIES
-    value: $(tasks.build-container.results.JAVA_COMMUNITY_DEPENDENCIES)
+    value: $(tasks.build-container-amd64.results.JAVA_COMMUNITY_DEPENDENCIES)
```
Comment (msugakov) [inline .tekton/collector-component-pipeline.yaml:105]: [not about this line] Please don't forget to adjust the comment in the `init` task: we don't use `apply-tags` any more but rather retag via building a manifest.

File: .tekton/collector-component-pipeline.yaml [hunk 2]
```diff
@@ -243,32 +243,195 @@ spec:
...
+        - name: bundle
+          value: quay.io/konflux-ci/tekton-catalog/task-build-image-manifest:0.1@sha256:fd0a0cf019621d6b577f1b9ab774bb1832f7cba61b4ceee2fd1bffc96895abf9
+        - name: kind
+          value: task
+      resolver: bundles
+    when:
+      - input: $(tasks.init.results.build)
+        operator: in
+        values:
+          - "true"
```
Comment (msugakov) [inline .tekton/collector-component-pipeline.yaml:397]: [nit]
```suggestion
        values: [ "true" ]
```

File: .tekton/collector-component-pipeline.yaml [hunk 3]
```diff
@@ -243,32 +243,195 @@ spec:
...
+
+  - name: build-image-manifest-extra
+    matrix:
+      params:
+        - name: IMAGE
+          value:
+            - $(params.output-image-repo):$(tasks.determine-image-tag.results.IMAGE_TAG)-latest
+            - $(params.output-image-repo):$(tasks.determine-image-tag.results.IMAGE_TAG)-slim
+            - $(params.output-image-repo):konflux-$(params.revision)
```
Comment (msugakov) [inline .tekton/collector-component-pipeline.yaml:406]: [no-op] Nice.

File: .tekton/collector-component-pipeline.yaml [hunk 4]
```diff
@@ -243,32 +243,195 @@ spec:
...
+        - name: bundle
+          value: quay.io/konflux-ci/tekton-catalog/task-build-image-manifest:0.1@sha256:fd0a0cf019621d6b577f1b9ab774bb1832f7cba61b4ceee2fd1bffc96895abf9
+        - name: kind
+          value: task
+      resolver: bundles
+    when:
+      - input: $(tasks.init.results.build)
+        operator: in
+        values:
+          - "true"
```
Comment (msugakov) [inline .tekton/collector-component-pipeline.yaml:429]:
```suggestion
        values: [ "true" ]
```

**Review by msugakov — COMMENTED** [review]
Minor comments. Please ping me for re-review after the build pipeline succeeds. I want to check pipeline outputs and glance at EC before approving.

**Review by msugakov — APPROVED** [review]

File: .tekton/collector-component-pipeline.yaml [hunk 5]
```diff
@@ -230,32 +230,199 @@ spec:
...
+        values: [ "true" ]
+
+  - name: build-image-manifest-extra
+    matrix:
+      params:
+        - name: IMAGE
+          value:
+            - $(params.output-image-repo):$(tasks.determine-image-tag.results.IMAGE_TAG)-latest
+            - $(params.output-image-repo):$(tasks.determine-image-tag.results.IMAGE_TAG)-slim
+            - $(params.output-image-repo):konflux-$(params.revision)
```
Comment (Molter73) [inline .tekton/collector-component-pipeline.yaml:398]: [nit] Didn't know this was a thing! We could've used a single manifest build for all components.

Anyways, I don't see this as a blocker, but maybe we could unify manifest creation on follow up PRs.

**Review by Molter73 — APPROVED** [review]
LGTM!

**Pull request merged**